integrated_channels/integrated_channel/tasks.py

```diff
@@ -89,6 +90,44 @@ def _log_batch_task_finish(task_name, channel_code, job_user_id,
         ))
 
 
+def set_enterprise_customer_site_to_request(integrated_channel):
+    """
+    Set the site of the current request to the site of the integrated channel's enterprise customer.
+
+    This is necessary since during the transmission of learner data, the request's site is used to determine
+    the domain for LMS API calls. This ensures that the correct site configuartions are used when making requests 
+    to the LMS.
+    """
+    request = get_current_request()
+    if request:
+        LOGGER.info(generate_formatted_log(
+            integrated_channel.channel_code(),
+            integrated_channel.enterprise_customer.uuid if integrated_channel.enterprise_customer else None,
+            None,
+            None,
+            f'Setting site to {integrated_channel.enterprise_customer.site.domain}'
+        ))
+        enterprise_customer = getattr(integrated_channel, 'enterprise_customer', None)
+        if not enterprise_customer:
+            LOGGER.warning(
+                'No enterprise customer found for integrated channel %s',
+                integrated_channel.channel_code()
+            )
+            return
+        site = getattr(enterprise_customer, 'site', None)
```

Comment:
A site is necessary to create an enterprise customer and the site is part of the attributes of the Enterprise customer class, so I don't think is necessary to check if the attribute exist, but to check if the attribute is valid. @alexjmpb